azurerm/helpers/azure/key_vault.go

```diff
@@ -118,3 +120,46 @@ func KeyVaultExists(ctx context.Context, client *keyvault.VaultsClient, keyVault
 
 	return true, nil
 }
+
+func KeyVaultCustomizeDiff(d *schema.ResourceDiff, _ interface{}) error {
+	if d.HasChange("soft_delete_enabled") {
+		if old, new := d.GetChange("soft_delete_enabled"); old.(bool) && !new.(bool) {
+			return fmt.Errorf("the property 'soft_delete_enabled' cannot be set to false, enabling the soft delete for a vault is an irreversible action")
+		}
+	}
+
+	if d.HasChange("purge_protection_enabled") {
+		if old, new := d.GetChange("purge_protection_enabled"); old.(bool) && !new.(bool) {
+			return fmt.Errorf("the property 'purge_protection_enabled' cannot be set to false, enabling the purge protection for a vault is an irreversible action")
+		}
+	}
+
+	return nil
+}
```

Comment:
(as mentioned offline) after spending some time testing this it appears this needs to come out - since if importing an existing key vault with soft delete/purge protection enabled, but the fields are omitted from the config - then this error will be shown, which is unclear how to proceed.

since we can move to using delta updates for this resource it should be safe to instead both add this to the documentation & return an error in the update method